### DISCUSSION TITLE
Next.js Auth.js v5 -- Error in build/production mode.

### DISCUSSION THREAD

**Discussion by AmeerMuaviya** [nextauthjs/next-auth#10264] [Help] [answered]
Hi. I'm using next.js ``` "next": "^14.0.3" ``` and ```  "next-auth": "^5.0.0-beta.13"```

**This is my `@/auth.ts` File**
```js
import NextAuth from "next-auth";
import { PrismaAdapter } from "@auth/prisma-adapter";
import authConfig from "./auth.config";
import db from "./components/utils/db";
export const {
  handlers: { GET, POST },
  auth,
  signIn,
  signOut,
} = NextAuth({
  secret: process.env.AUTH_SECRET,
  adapter: PrismaAdapter(db),
  session:{strategy:"jwt"},
  ...authConfig,
  trustHost: true,
  basePath:"/api/auth"
});
```

**Here is the `@/auth.config.ts`**
```js
import CredentialsProvider from "next-auth/providers/credentials";
import type { NextAuthConfig } from "next-auth";
import { loginSchema } from "./schema/login-schema";
import GoogleProvider from "next-auth/providers/google";
import db from "./components/utils/db";
import bcrypt from "bcryptjs";
export default {
  providers: [
    CredentialsProvider({
      async authorize(credentials) {
        const validatedFields = loginSchema.safeParse(credentials);
        if (validatedFields.success) {
          const { username, password } = validatedFields.data;
          const user = await db.user.findUnique({
            where: {
              username,
            },
          });
          if (!user || !user.password) {
            return null;
          }
          const passwordMatch = await bcrypt.compare(password, user.password);
          if (passwordMatch) {
            return user;
          }
        }
        return null;
      },
    }),
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
    }),
  ],
} satisfies NextAuthConfig;
```
**I also have a `@/middleware.tsx`**
```js
import authConfig from "./auth.config";
import { apiAuthPrefix, authRoutes, defaultLoginRedirect } from "@/routes";
import NextAuth from "next-auth";

const { auth } = NextAuth(authConfig);
export default auth((req) => {
  const isLoggedIn = !!req.auth;
  console.log('isLoggedIn: ', isLoggedIn);
  const { nextUrl } = req;
  const isApiAuthRoute = nextUrl.pathname.startsWith(apiAuthPrefix);
  const isAuthRoute = authRoutes.includes(nextUrl.pathname);
  if (isApiAuthRoute) {
    return;
  }
  if (isAuthRoute) {
    if (isLoggedIn) {
      return Response.redirect(new URL(defaultLoginRedirect, nextUrl));
    }
    return;
  }
  if (!isLoggedIn) {
    return Response.redirect(new URL("/auth/login", nextUrl));
  }
});

export const config = {
  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
};
```

```env
DATABASE_URL="******"

NEXTAUTH_URL=http://localhost:3000/api/auth

AUTH_SECRET=******************
GOOGLE_CLIENT_ID=***************
GOOGLE_CLIENT_SECRET=************
```
With this credentials and google auth are working fine. but when i run the command 
```sh
yarn build && yarn start
```

it gives me the following error
```error
TypeError: iB is not a constructor
    at iV (/home/user/Coding/project/.next/server/chunks/341.js:373:50596)
    at t (/home/user/Coding/project/.next/server/chunks/341.js:373:56431)
    at /home/user/Coding/project/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:41328
```
This is the `/home/user/Coding/project/.next/server/chunks/341.js:373:50596` file where the error is accuring
```js
 var iB = r(80428);
      function iV(e) {
        let t = process.env.AUTH_URL ?? process.env.NEXTAUTH_URL;
        if (!t) return e;
        let { origin: r } = new URL(t),
          { href: n, origin: i } = e.nextUrl;
        return new iB(n.replace(i, r), e);
      }
```
And these errors in Devtools. 
![Screenshot from 2024-03-09 10-30-01](https://github.com/nextauthjs/next-auth/assets/88961101/7af0c939-d88a-42f9-a85c-6c5b22054acf)

I searched for solutions online. 
### I've tried
Removing `NEXTAUTH_URL` from `.env` file but it gives me `[auth][error] UnknownAction: Cannot parse action at /session`. 
Sometimes it also says `Untrusted host error`

### I'm unable to

1. Login using google. it redirects me to `/api/auth/error`
2. Logout from the app

Please help me to fix this issue.

**Comment by AmeerMuaviya** [comment, marked as answer]
setting `AUTH_TRUST_HOST="true"` in `.env` fixed my issue. 🥲

**Reply by timaxlucas** [reply]
I am having the same problem in the same environment. Your solution, however, did not fix the issue for me. Did you change anything else?

EDIT: I don't get the error when omitting`NEXTAUTH_URL` and including `NEXTAUTH_URL_INTERNAL`. But my setup is behind a proxy sothe "Sign in with Google" form has wrong internal links.

**Comment by Apezdr** [comment]
I'm having this issue after updating, getting the `[auth][error] UnknownAction: Cannot parse action at /session` error message and no hints as to what is broken setting Trust Host isn't working but I'm not getting host errors. Anything from the team?

**Reply by timaxlucas** [reply]
I had the same issue. Upgrading to next `14.1.2` and setting `basePath` in NextAuth config and setting **full** url in `NEXTAUTH_URL` including `/api/auth` fixed it for me.

Are you experiencing this locally or after deployed?

**Reply by timaxlucas** [reply]
When running locally: remove the `AUTH_URL` / `NEXTAUTH_URL` from env. It's only really needed when self hosted behind a proxy.

**Reply by timaxlucas** [reply]
see https://github.com/nextauthjs/next-auth/issues/10081
